Fix issue when try to display v4 APIs in health-check page

michel-barret opened the pull request:
## Issue

https://gravitee.atlassian.net/browse/APIM-7865

## Description

- **refacto: some cleanup**
- **feat: use mapping**
- **feat: accept all type of APIs**
<!-- Storybook placeholder -->
---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-gdzchrwjqg.chromatic.com)
<!-- Storybook placeholder end -->

michel-barret committed:
refacto: some cleanup

APIM-7865

michel-barret committed:
feat: update PlanServiceImpl to handle v4

Use mapping instead of cast to handle all types of APIs in PlanServiceImpl

APIM-7865

jgiovaresco commented on gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/PlanServiceImpl.java:
```diff
@@ -149,30 +151,29 @@ public class PlanServiceImpl extends AbstractService implements PlanService {
 
             if (api.getApiLifecycleState() == DEPRECATED) {
-                throw new ApiDeprecatedException(api.getName());
+                throw new ApiDeprecatedException(api.getId());
```
I understand the change but I thing it would have been better to isolate that in a dedicated commit because it has a functional impact

jgiovaresco commented on gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/PlanServiceImpl.java:
```diff
@@ -149,30 +151,29 @@ public class PlanServiceImpl extends AbstractService implements PlanService {
             validateTags(newPlan.getTags(), api);
 
-            String id = newPlan.getId() != null && UUID.fromString(newPlan.getId()) != null ? newPlan.getId() : UuidString.generateRandom();
+            String id = newPlan.getId() != null ? newPlan.getId() : UuidString.generateRandom();
```
I guess you check that the result is the same but I don't really understand by we were encapsulating with a `fromString` before 🙈

michel-barret committed:
feat: accept all type of APIs

Update heath check page to accept all type of API.

I also use signal to fix async issue (in some case the update of state was ignored).

APIM-7865
